### PR TITLE
ODPR-1632[AK]- Added notification Journey

### PR DESCRIPTION
**Pull request by AKKUM**

### CODE REVIEW

**Review by georgievh — APPROVED** [review]

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      resultPage.heading shouldBe "Manage your digital platform reporting"
+    }
+
+    Scenario("Add operator notification with EPO Notification where we have multiple operators") {
+      Given("Organisation user is subscribed")
+      val enrolmentsData      = SubscriptionSteps.subscribedOrganisationEnrolment()
+      And("Organisation user logs in")
+      loginPage.show()
+      loginPage.loginAs(aUserCredentials.copy(enrolmentsData = enrolmentsData))
+      And("Already added the platform operator")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "Already added **more than one** platform operator"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+class AddAndViewReportingNotificationJourneysSpec extends BaseSpec {
+
+  private val loginPage                 = AuthLoginStubPage
+  private val indexPage                 = IndexPage()
+  private val resultPage                = ResultPage
+  private val whichPlatformOperatorPage = WhichPlatformOperatorPage()
+
+  Feature("Add Operator Notification Journeys") {
+    Scenario("Add operator notification with RPO Notification for extended and active seller due diligence") {
+      Given("Organisation user is subscribed")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user is subscribed"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      val enrolmentsData = SubscriptionSteps.subscribedOrganisationEnrolment()
+
+      And("Organisation user logs in")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user logs in"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      NotificationCheckYourAnswersPage(platformOperatorId).continue()
+      NotificationSuccessPage(platformOperatorId).clickManageYourDigitalPlatformReporting()
+
+      Then("The result page should be 'Manage your digital platform reporting'")
+      resultPage.url       should include("/manage-reporting")
+      resultPage.heading shouldBe "Manage your digital platform reporting"
+    }
+
+    Scenario("Add operator notification with EPO Notification") {
+      Given("Organisation user is subscribed")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user is subscribed"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      val enrolmentsData     = SubscriptionSteps.subscribedOrganisationEnrolment()
+      And("Organisation user logs in")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user logs in"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+      NotificationCheckYourAnswersPage(platformOperatorId).continue()
+      NotificationSuccessPage(platformOperatorId).clickManageYourDigitalPlatformReporting()
+
+      Then("The result page should be 'Manage your digital platform reporting'")
+      resultPage.url       should include("/manage-reporting")
+      resultPage.heading shouldBe "Manage your digital platform reporting"
+    }
+
+    Scenario("Add multiple operator notification with one RPO and one EPO Notification") {
+      Given("Organisation user is subscribed")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user is subscribed"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+      val enrolmentsData     = SubscriptionSteps.subscribedOrganisationEnrolment()
+      And("Organisation user logs in")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user logs in"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+      whichPlatformOperatorPage.continue()
+      NotificationViewPage(platformOperatorId).selectNo().continue()
+
+      Then("The result page should be 'Manage your digital platform reporting'")
+      resultPage.url       should include("/manage-reporting")
+      resultPage.heading shouldBe "Manage your digital platform reporting"
+    }
+
+    Scenario("View multiple operator notification with one RPO and one EPO Notification") {
+      Given("Organisation user is subscribed")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user is subscribed"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+      val enrolmentsData     = SubscriptionSteps.subscribedOrganisationEnrolment()
+      And("Organisation user logs in")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user logs in"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+      ManageReportingPage(platformOperatorId).clickViewReportingNotification()
+      NotificationViewPage(platformOperatorId).selectNo().continue()
+
+      Then("The result page should be 'Manage your digital platform reporting'")
+      resultPage.url       should include("/manage-reporting")
+      resultPage.heading shouldBe "Manage your digital platform reporting"
+    }
+
+    Scenario("Add operator notification with EPO Notification where we have multiple operators") {
+      Given("Organisation user is subscribed")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user is subscribed"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+      val enrolmentsData      = SubscriptionSteps.subscribedOrganisationEnrolment()
+      And("Organisation user logs in")
```
Comment (jassalrichy) [inline src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: "**User** user logs in"

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 1 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala:174]: Here we only one platform operator. So text is fine.

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 2 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 3 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 4 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 5 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 6 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 7 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 8 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 9 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 10 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 11 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala (hunk 1 above)
Comment (AKKUM) [reply src/test/scala/uk/gov/hmrc/ui/specs/reportingnotifications/AddAndViewReportingNotificationJourneysSpec.scala]: Done

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**